packages/training/book/src/training-pipeline/trajectory-format.md

```
@@ -0,0 +1,348 @@
+    if not has_llm:
+        issues.append("No LLM calls found")
+    
+    return len(issues) == 0, issues
+```
+
+## JSON Mode vs DB Mode
+
+### JSON Mode (Development)
+
+Trajectories saved to files:
+```
```

markdownlint-cli2 (0.18.1)

Fenced code blocks should be surrounded by blank lines

(MD031, blanks-around-fences)